src/plugins/expressions/public/loader.ts

```diff
@@ -46,12 +46,20 @@ export class ExpressionLoader {
     params?: IExpressionLoaderParams
   ) {
     this.dataSubject = new Subject();
-    this.data$ = this.dataSubject.asObservable().pipe(share());
-
-    this.loadingSubject = new Subject();
-    this.loading$ = this.loadingSubject.asObservable().pipe(share());
-
-    this.renderHandler = new ExpressionRenderHandler(element);
+    this.data$ = this.dataSubject.asObservable();
+
+    this.loadingSubject = new BehaviorSubject<boolean>(false);
+    // loading is a "hot" observable,
+    // as loading$ could emit straight away in the constructor
+    // and we want to notify subscribers about it, but all subscriptions will happen later
+    this.loading$ = this.loadingSubject.asObservable().pipe(
+      filter(_ => _ === true),
```

Comment:
can you please explain why we need boolean observable, where we then filter out all the falses and map trues to void ?

Reply:
The intention was to leave public api (loading$) unchanged in this pr and just fix the bug where `expression_renderer.tsx` was missing initial loading state. This was fixed by changing Subject to BehaviourSubject to remember last emitted value